src/res/KM_ResKeys.pas

```diff
@@ -0,0 +1,287 @@
+
+implementation
+
+{ TKMKeyLibraryMulti }
+constructor TKMKeyLibraryMulti.Create;
+begin
+  inherited Create;
+end;
+
+
+// .keymap files consist of lines. Each line has an index and a text. Lines without index are skipped
+procedure TKMKeyLibraryMulti.LoadKeymapFile(var aArray: TIntegerArray);
+  function KeyToArray(const Value: UnicodeString): TUnicodeStringArray;
+  var
+    P, Start: PWideChar;
+    S: UnicodeString;
+  begin
+    SetLength(Result, 0);
+    P := Pointer(Value);
+    if P = nil then Exit;
+
+    // This is a lot faster than using StrPos/AnsiStrPos when
+    // LineBreak is the default (#13#10)
+    while P^ <> #0 do
+    begin
+      Start := P;
+      while not KromUtils.CharInSet(P^, [#0, #10, #13]) do Inc(P);
+      SetString(S, Start, P - Start);
+
+      SetLength(Result, Length(Result) + 1);
+      Result[High(Result)] := S;
+
+      if P^ = #13 then Inc(P);
+      if P^ = #10 then Inc(P);
+    end;
+  end;
+var
+  Tmp: TUnicodeStringArray;
+  s, libKey: UnicodeString;
+  I, id, topId, firstDelimiter: Integer;
+  {$IFDEF FPC} tmpA: AnsiString; {$ENDIF}
+begin
+  FILE_PATH := (ExeDir + 'data' + PathDelim + 'keys.keymap');
+  if not FileExists(FILE_PATH) then Exit;
+
+  // Load UniCode file with codepage
+  libKey := ReadTextU(FILE_PATH, 1252);
+  Tmp := KeyToArray(libKey);
+  fKeyCount := High(Tmp);
+
+  for I := High(Tmp) downto 0 do
+  begin
+    firstDelimiter := Pos(':', Tmp[I]);
+    if firstDelimiter = 0 then Continue;
+
+    if TryStrToInt(LeftStr(Tmp[I], firstDelimiter - 1), topId) then
+      Break;
+  end;
+
+  Assert(topId <= 1024, 'Dont allow too many strings for no reason');
+
+  // Don't shrink the array, we might be overloading
+  if Length(aArray) < topId + 1 then
+    SetLength(aArray, topId + 1);
+
+  for I := 0 to High(Tmp) do
+  begin
+    s := Tmp[I];
+
+    // Get Key index and skip erroneous lines
+    firstDelimiter := Pos(':', s);
+    if firstDelimiter = 0 then Continue;
+    if not TryStrToInt(TrimLeft(LeftStr(s, firstDelimiter - 1)), id) then Continue;
+
+    s := RightStr(s, Length(s) - firstDelimiter);
+    // Required characters that can't be stored in plain text
+    //todo: Remove them in favor of | for eol (and update libx files)
+    {$IFDEF WDC}
+    s := StringReplace(s, '\n', EolW, [rfReplaceAll, rfIgnoreCase]); // EOL
+    s := StringReplace(s, '\\', '\', [rfReplaceAll, rfIgnoreCase]); // Slash
+    {$ENDIF}
+    {$IFDEF FPC}
+    // In FPC StringReplace only works for UTF8/Ansi strings
+    tmpA := UTF16toUTF8(s);
+    tmpA := StringReplace(tmpA, '\n', EolW, [rfReplaceAll, rfIgnoreCase]); // EOL
+    tmpA := StringReplace(tmpA, '\\', '\', [rfReplaceAll, rfIgnoreCase]); // Slash
+    s := UTF8toUTF16(tmpA);
+    {$ENDIF}
+    aArray[id] := StrToInt(s);
+  end;
+end;
+
+
+// Check if requested string is empty
+function TKMKeyLibraryMulti.HasKey(aIndex: Word): Boolean;
+begin
+  Result := ((0 <> -1) and (aIndex < Length(fKeys)) and (fKeys[aIndex] <> 0));
```

Comment:
Removed. :D 
Somehow it gave me an error before, no idea why.
(0 <> -1) and  has completely been removed now, I face-palmed over it quite massively.